### PR TITLE
Update bootstrap researchplans

### PR DESCRIPTION
**Pull request by ashwiniHerle**
This PR covers react-bootstrap upgrade for the components under:

- [x] apps/mydb/elements/details/researchPlans/*.js:

     researchPlanTab/*
     analysesTab/*
     attachmentsTab/*

- [x] components/structureEditor/StructureEditorModal.js

 **Components under apps/mydb/elements/details/researchPlans/wellPlatesTab are not covered in this PR**

### CODE REVIEW

File: app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsField.js [hunk 1]
```diff
@@ -127,53 +138,59 @@ export default class ResearchPlanDetailsField extends Component {
...
+                        {element.title}
+                      </Dropdown.Item>
+                    ))
+                  }
+                </Dropdown.Menu>
+              </Dropdown>
             </ButtonGroup>
           </OverlayTrigger>
         );
       }
```
Comment (beque) [inline app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsField.js:168]: here are some pull-right classes in this file

File: app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldImage.js [hunk 2]
```diff
@@ -60,38 +65,43 @@ export default class ResearchPlanDetailsFieldImage extends Component {
...
+          />
         </div>
       );
     } else {
       content = <p>Drop Files, or Click to Select.</p>;
     }
     return (
       <div>
-        <FormGroup style={{ width: '30%' }}>
+        <Form.Group style={{ width: '30%' }}>
```
Comment (beque) [inline app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldImage.js:96]: Maybe you can try the col classes for the width
https://getbootstrap.com/docs/5.3/layout/grid/

File: app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldTableMeasurementExportModal.js [hunk 3]
```diff
@@ -154,7 +154,7 @@ class ResearchPlanDetailsFieldTableMeasurementExportModal extends Component {
           </Modal.Title>
         </Modal.Header>
         <Modal.Body className='measurementExportModal__body'>
-          <table className="table">
+          <Table className="table">
```
Comment (beque) [inline app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldTableMeasurementExportModal.js:157]: Is this class "table" necessary?

File: app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldTableMeasurementExportModal.js [hunk 4]
```diff
@@ -164,35 +164,33 @@ class ResearchPlanDetailsFieldTableMeasurementExportModal extends Component {
...
+                  value={candidate.value}
+                />
+              ))}
             </tbody>
-          </table>
-          <div>
-            <ButtonToolbar>
+          </Table>
+        </Modal.Body>
+        <Modal.Footer className="modal-footer border-0">
```
Comment (beque) [inline app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldTableMeasurementExportModal.js:184]: Is this class "modal-footer" necessary?

File: app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsName.js [hunk 5]
```diff
@@ -18,21 +17,12 @@ export default class ResearchPlanDetailsName extends Component {
       >
         <Button
           id="copyMetadataButton"
-          title=""
           className="fa fa-laptop pull-right"
```
Comment (beque) [inline app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsName.js:20]: Another pull-right class

File: app/assets/stylesheets/research-plan.scss [hunk 6]
```diff
@@ -197,4 +197,3 @@
         display: none
     }
 }
-
```
Comment (beque) [inline app/assets/stylesheets/research-plan.scss:200]: You have removed many classes in the components.
Are there any classes left from this file?
If not, you could delete this file

**Review by beque — COMMENTED** [review]
Just some minor fixes ;-)

File: app/assets/stylesheets/chemotion_bootstrap.scss [hunk 7]
```diff
@@ -106,5 +106,6 @@ $theme-colors: map-merge($theme-colors, $gray-tones);
 }
 
 .border-dashed {
-  border-style: dashed;
+  border: dashed;
```
[the remark this answers is not in the document]
Comment (ashwiniHerle) [reply app/assets/stylesheets/chemotion_bootstrap.scss:109]: For some reason, it did not work with just border-dashed class. I will look into it again

File: app/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanMetadata.js [hunk 8]
```diff
@@ -169,352 +162,351 @@ export default class ResearchPlanMetadata extends Component {
...
+                      type="text"
+                      value={description?.descriptionType}
+                      placeholder="Type"
+                      onChange={event => this.updateResearchPlanMetadataArrayItem('description', index, 'descriptionType', event.target.value)}
+                    />
+                  </Form.Group>
+                </Col>
+                <Col smOffset={0} sm={2}>
+                  <Form.Label>Action</Form.Label><br />
+                  <Button variant="danger" className="pull-right" size="sm" onClick={() => this.removeResearchPlanMetadataArrayItem('description', index)}>
```
[the remark this answers is not in the document]
Comment (ashwiniHerle) [reply app/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanMetadata.js:300]: Yes, it still works. I found few places where normal justify-content and similar classes didn't work and pull-right has worked.

File: app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldTableMeasurementExportModal.js (hunk 3 above)
Comment (ashwiniHerle) [reply app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldTableMeasurementExportModal.js:157]: Probably not. Will make necessary changes

**Review by maiwald — COMMENTED** [review]
Some minor notes 😊

File: app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldTable.js [hunk 9]
```diff
@@ -161,7 +154,7 @@ export default class ResearchPlanDetailsFieldTable extends Component {
     onChange(field.value, field.id);
   }
 
-  handleSchemaModalShow() {
+  handleSchemaModalShow = () => {
```
[the remark this answers is not in the document]
Comment (ashwiniHerle) [reply app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldTable.js:157]: We use .bind in many places. Got rid of the .bind after using arrow function

File: app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldImage.js [hunk 10]
```diff
@@ -113,13 +113,13 @@ export default class ResearchPlanDetailsFieldImage extends Component {
...
           </InputGroup>
         </Form.Group>
-        <SaveEditedImageWarning visible={is_annotationUpdated} />
         <Dropzone
           accept="image/*"
           multiple={false}
           onDrop={(files) => this.handleDrop(files)}
           className="border-dashed border-gray-300 text-center p-2 mb-3"
         >
+          {isAnnotationUpdated && <SaveEditedImageWarning visible />}
```
[the remark this answers is not in the document]
Comment (ashwiniHerle) [reply app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldImage.js:120]: It is used in other components as of now

File: app/packs/src/apps/mydb/elements/details/researchPlans/analysesTab/ResearchPlanDetailsContainers.js [hunk 11]
```diff
@@ -181,7 +181,7 @@ export default class ResearchPlanDetailsContainers extends Component {
       const contentOneLine = {
         ops: content.ops.map((x) => {
           const c = Object.assign({}, x);
-          if (c.insert) c.insert = truncateText(c.insert.replace(/\n/g, ' '), 300);
+          if (c.insert) c.insert = truncateText(c.insert.replace(/\n/g, ' '), 100);
```
[the remark this answers is not in the document]
Comment (beque) [reply app/packs/src/apps/mydb/elements/details/researchPlans/analysesTab/ResearchPlanDetailsContainers.js:184]: Maybe you can use this:
https://getbootstrap.com/docs/5.3/helpers/text-truncation/

[the remark this answers is not in the document]
Comment (ashwiniHerle) [reply app/packs/src/apps/mydb/elements/details/researchPlans/analysesTab/ResearchPlanDetailsContainers.js:184]: Done. Moved to textHelper and reused it.

**Pull request merged**